pkg/collateral/control.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collateral
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+)
+
+// Control determines the behavior of the EmitCollateral function
+type Control struct {
+	// OutputDir specifies the directory to output the collateral files
+	OutputDir string
+
+	// EmitManPages controls whether to produce man pages.
+	EmitManPages bool
```

Comment:
Just curious ... is there a use case for doing more than one operation? Or would the user typically only want to emit one form?  I'm wondering if it should be a single enum rather than booleans.

Reply:
Copying reply from @geeknoid here (to retain the thread):

> I have an "all" option on the command-line to output everything. In
reality, I don't expect anyone will actually use the EmitCollateral
function outside of this package. I really only exposed it for anal
retentive/OCD reasons :-)

Reply:
@geeknoid ah ok :) ... the reason I was asking is that my immediate assumption was that the default output would be `stdout`.  But it seems that the default is to dump files in the current working directory. Would `stdout` ever be desirable for any of these outputs ... would it even make sense?

Reply:
Multiple files are produced for each kind of output, except bash. So no, I don't think stdout is useful.

Let's not overthink this. This feature will be used by the doc extraction process in istio.github.io and that's about it. It's unlikely anybody will ever need to use this outside of that specific case.

Reply:
Yup makes sense ... thanks.